Source/Task/StaticArray.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+template <typename TData, uint32_t size>
+class StaticArray
```

Comment:
what does this give us that std::array does not?

Reply:
It has some convenience -- you can remove an element from the middle and the contents shift.  Basically, I can treat this like a vector or other container from an API standpoint, but it's fixed size with no allocation.

Reply:
we could build those convenience functions on top of std::array, not that this code is that tricky, but doesn't feel right to rewrite it

Reply:
I looked at this a little deeper.  I don't think there's value in using std::array for this.  The only thing I really get from std::array is a [] operator.  The rest of the api in StaticArray would need to be added to std::array through derivation (you still are defnining a new class) or helper methods (which removes the ability to treat this like an object).

Reply:
ok